src/Microsoft.AspNet.Mvc.Core/ModelBinding/CompositeModelBinder.cs

```diff
@@ -39,6 +40,12 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
                 !string.IsNullOrEmpty(bindingContext.ModelName);
 
             var newBindingContext = CreateNewBindingContext(bindingContext, bindingContext.ModelName);
+            if (newBindingContext == null)
```

Comment:
Basically this and `MutableObjectModelBinder` treated all `IBindingSourceValueProvider` implementations as if they were guaranteed to act like `CompositeValueProvider`.